state-chain/runtime/src/chainflip/bitcoin_block_processor.rs

```diff
@@ -235,6 +229,23 @@ impl Hook<HookTypeFor<TypesEgressWitnessing, SafetyMarginHook>> for TypesEgressW
 
 #[cfg(test)]
 mod tests {
+	use crate::chainflip::bitcoin_block_processor::{dedup_events, BtcEvent};
+
+	#[test]
+	fn dedup_events_test() {
+		let events = vec![
+			(10, BtcEvent::<u8>::Witness(9)),
+			(8, BtcEvent::<u8>::PreWitness(9)),
+			(10, BtcEvent::<u8>::Witness(10)),
+			(10, BtcEvent::<u8>::Witness(11)),
+			(8, BtcEvent::<u8>::PreWitness(11)),
+			(10, BtcEvent::<u8>::PreWitness(12)),
+		];
+		let deduped_events = dedup_events(events);
+		assert_eq!(deduped_events.len(), 4);
+		assert!(!deduped_events.contains(&(8, BtcEvent::<u8>::PreWitness(9))));
+		assert!(!deduped_events.contains(&(8, BtcEvent::<u8>::PreWitness(11))));
```

Comment:
I think it would be useful to check against the full result vector here (that we expect), then the expected behavior is clearer on a quick glance & we're actually checking more.